portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,27 +40,34 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+        //Get an instance of datastore and prepare a query of the comments sorted by date:
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
     
+        //Query the datastore and get the result with the comments:
         PreparedQuery results = datastore.prepare(query);
 
+        //Create and ArrayList that stores Comment objects created with the properties form the entities that where queried from the datastore:
         ArrayList<Comment> comments = new ArrayList<Comment>();
         for(Entity entity : results.asIterable()){
 
             String username = (String) entity.getProperty("username");
             String message = (String) entity.getProperty("message");
             String date = (String) entity.getProperty("date");
+            String emoji = (String) entity.getProperty("emoji");
+            double score = (double) entity.getProperty("score");
 
-            Comment current = new Comment(username, message, date);
+            Comment current = new Comment(username, message, date, score, emoji);
 
             comments.add(current);
 
         }
 
+        //Create a JSON from the ArrayList and send it as a response:
         String json = toJsonString(comments);
 
-        response.setContentType("application/json;");
+        //It needs to specify its encoded in UTF-8 so the JSON parcer in the client can recognize the emojis:
```

Comment:
parser*

Great comment explaining `charset=utf-8`. Without the comment, other engineers working with this code may be tempted to remove it if they don't think it's needed.

Reply:
Thanks! I will correct my spelling.